shell/plugins/dashboard-store/mutations.js

```diff
@@ -165,6 +180,71 @@ export function remove(state, obj, getters) {
   }
 }
 
+export function batchChanges(state, { ctx, batch }) {
+  const types = Object.keys(batch);
+
+  types.forEach((type) => {
+    const normalizedType = normalizeType(type === 'counts' ? COUNT : type);
+    const keyField = keyFieldFor(normalizedType);
+    const typeCache = registerType(state, normalizedType);
+    const opts = ctx.rootGetters[`type-map/optionsFor`](type);
+    const limit = opts?.limit;
+
+    const createdResources = [];
+    const batchActionMap = Object.keys(batch[normalizedType]).map((id) => {
+      const index = typeCache.list.findIndex((resource) => {
+        return resource[keyField] === id;
+      });
+      const resource = batch[normalizedType][id];
+      let action = 'change';
+
+      if (resource.remove) {
+        action = 'remove';
+      } else if (index === -1) {
+        action = 'create';
+        createdResources.push(specialFields(normalizedType, resource));
+      }
+
+      return {
+        id,
+        index,
+        resource: action === 'change' ? specialFields(normalizedType, resource) : undefined,
+        action
+      };
+    }).reduce((acc, batchIdMap) => {
+      return {
+        ...acc,
+        [batchIdMap[keyField]]: batchIdMap
+      };
+    }, {});
+
+    const newCache = typeCache.list.reduce((acc, resource) => {
+      const resourceId = resource[keyField];
+      const batchAction = batchActionMap[resourceId];
+
+      if (!batchAction) { // use the unchanged resource from the cache
+        return acc.concat(resource);
+      } else if (batchAction.action === 'change') { // use the changed resource from the map
+        return acc.concat(batchAction.resource);
+      }
+
+      return acc;
+    }, [])
+      .concat(createdResources)
+      .slice(limit ? -limit : undefined)
+      .map(resource => classify(ctx, resource));
```

Comment:
that makes sense, I was classifying again just to make sure I also classified the added resources as well, seemed relatively cheap since the function actually returns immediately if the object's type is already a resource but it's just as easy to classify the created resources when I push them into their array.